### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by sophiasph**
- [DEMO LINK](https://sophiasph.github.io/layout_catalog/)
- [TEST REPORT LINK](https://sophiasph.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,417 @@
...
-    <h1>Catalog</h1>
+    <header class="header">
+      <a
+      href="#"
+      class="header__logo"
+      >
+        <img
+        src="./images/logo.png"
+        alt="logo__MOYO"
+        >
```
Comment (Esceype) [inline src/index.html:18]:
```suggestion
        <img
          src="./images/logo.png"
          alt="logo__MOYO"
        >
```

Add two more spaces for attributes if they are in new lines

File: src/index.html [hunk 2]
```diff
@@ -7,6 +7,417 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Catalog</h1>
+    <header class="header">
+      <a
+      href="#"
+      class="header__logo"
```
Comment (Esceype) [inline src/index.html:13]:
```suggestion
        href="#"
        class="header__logo"
```

File: src/index.html [hunk 3]
```diff
@@ -7,6 +7,417 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
```
Comment (Esceype) [inline src/index.html:8]: Import fonts

File: src/index.html [hunk 4]
```diff
@@ -7,6 +7,417 @@
...
+      href="#"
+      class="header__logo"
+      >
+        <img
+        src="./images/logo.png"
+        alt="logo__MOYO"
+        >
+      </a>
+
+      <nav class="header__nav">
```
Comment (Esceype) [inline src/index.html:21]: Make nav a separate BEM block, and not like header element, if you need to add some extra CSS props like indents use mix classes according to BEM.

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,239 @@
...
   margin: 0;
+  box-sizing: border-box;
+  font-family: Roboto, sans-serif;
+  font-size: 10px;
+  color: $--main-color-text;
+  font-weight: 400;
+}
+
+/* #region header */
+.header {
```
Comment (Esceype) [inline src/styles/main.scss:17]: To separate file as it is a separate BEM block

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,239 @@
...
+    border-radius: 8px;
+    position: absolute;
+    bottom: 0;
+    left: 0;
+  }
+}
+/* #endregion */
+
+/* #region cards */
+.card {
```
Comment (Esceype) [inline src/styles/main.scss:91]: Move it separate file too

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,239 @@
...
+  }
+
+  &__rating {
+    height: 16px;
+    display: flex;
+    justify-content: space-between;
+    align-items: flex-end;
+    margin-bottom: 24px;
+
+    &-stars {
```
Comment (Esceype) [inline src/styles/main.scss:113]: Stars should be separate block and have own file

File: src/styles/main.scss [hunk 8]
```diff
@@ -1,3 +1,239 @@
+@import "./variables.scss";
+
+* {
+  box-sizing: border-box;
+}
+
 body {
   margin: 0;
+  box-sizing: border-box;
```
Comment (Esceype) [inline src/styles/main.scss:9]: Remove it you already have this prop in * selector

File: src/styles/main.scss [hunk 9]
```diff
@@ -1,3 +1,239 @@
...
+  font-family: Roboto, sans-serif;
+  font-size: 10px;
+  color: $--main-color-text;
+  font-weight: 400;
```
Comment (Esceype) [inline src/styles/main.scss:13]: This is default value

File: src/styles/main.scss [hunk 10]
```diff
@@ -1,3 +1,239 @@
+@import "./variables.scss";
+
+* {
+  box-sizing: border-box;
+}
+
 body {
   margin: 0;
+  box-sizing: border-box;
+  font-family: Roboto, sans-serif;
```
Comment (Esceype) [inline src/styles/main.scss:10]: Add one more font as fallback font

File: src/styles/main.scss [hunk 11]
```diff
@@ -1,3 +1,239 @@
...
+  flex-direction: column;
+  padding: 32px 16px 16px;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: #fff;
+  transition-duration: 300ms;
+
+  &:hover {
+    width: calc(200px + 20%);
+  }
```
Comment (Esceype) [inline src/styles/main.scss:104]: It should scale, not just increase the width.

File: src/styles/main.scss [hunk 12]
```diff
@@ -1,3 +1,239 @@
...
+    width: 40px;
+    height: 40px;
+  }
+
+  &__nav {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &-ul {
```
Comment (Esceype) [inline src/styles/main.scss:33]: Rename it to list and use `__` instead of `-`

File: src/styles/main.scss [hunk 13]
```diff
@@ -1,3 +1,239 @@
...
+      list-style: none;
+      padding: 0;
+      margin: 0;
+    }
+
+    &-li {
```
Comment (Esceype) [inline src/styles/main.scss:39]: Rename it to the item and use double underline instead `-`

**Review by Esceype — CHANGES_REQUESTED** [review]
GJ!
But your tests report link doesn`t work. Also I don`t see that you import font Roboto, that`s why your text at page looks wrong.

Read the following comment in code.

File: src/index.html [hunk 14]
```diff
@@ -5,8 +5,422 @@
...
+
+      <nav class="nav">
+        <ul class="nav__ul">
+          <li class="nav__li">
+            <a
+              href="#"
+              class="nav-link is-active"
+            >
+              Apple
+            </a>
```
Comment (volodymyr-soltys97) [inline src/index.html:32]: Let's fix classes everywhere
```suggestion
      <nav class="nav">
        <ul class="nav__list">
          <li class="nav__item">
            <a
              href="#"
              class="nav__link is-active"
            >
              Apple
            </a>
```

File: src/index.html [hunk 15]
```diff
@@ -5,8 +5,422 @@
...
+          Product code: 195434
+        </p>
+
+        <div class="card__rating">
+          <div class="card__rating-stars card__rating-stars--5">
+            <div class="card__rating-stars-star active"></div>
+            <div class="card__rating-stars-star active"></div>
+            <div class="card__rating-stars-star active"></div>
+            <div class="card__rating-stars-star active"></div>
+            <div class="card__rating-stars-star active"></div>
```
Comment (volodymyr-soltys97) [inline src/index.html:161]: Redundant, you can just remove it everywhere
```suggestion
            <div class="card__rating-stars-star"></div>
            <div class="card__rating-stars-star"></div>
            <div class="card__rating-stars-star"></div>
            <div class="card__rating-stars-star"></div>
            <div class="card__rating-stars-star"></div>
```

File: src/index.html [hunk 16]
```diff
@@ -5,8 +5,422 @@
...
+        <div class="card__price">
+          <p class="card__price-text">Price:</p>
+          <p class="card__price-number">$2,199</p>
+        </div>
+
+        <button
+          class="card__button"
+        >
+          Buy
+        </button>
```
Comment (volodymyr-soltys97) [inline src/index.html:176]:
```suggestion
        <button class="card__button">
          Buy
        </button>
```

File: src/styles/cards.scss [hunk 17]
```diff
@@ -0,0 +1,101 @@
+@import "./stars";
+
+.card {
+  width: 200px;
+  height: 408px;
```
Comment (volodymyr-soltys97) [inline src/styles/cards.scss:5]: No need to hardcode height for the card, it will be better if the height depends on the content

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

File: src/index.html (hunk 15 above)
Comment (anastasiiavorobiova) [reply src/index.html:161]: Not fixed. Modifiers don't work 
Screenshot 2023-11-19 at 16 02 46

File: src/styles/cards.scss [hunk 18]
```diff
@@ -0,0 +1,99 @@
+@import "./stars";
+
+.card {
+  display: flex;
+  flex-direction: column;
+  padding: 32px 16px 16px;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: #fff;
```
Comment (anastasiiavorobiova) [inline src/styles/cards.scss:9]: Consider using variables for repeating colors

File: src/styles/cards.scss [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+    width: 100%;
+    font-size: 14px;
+    font-weight: 700;
+    line-height: 16px;
+    text-transform: uppercase;
+    padding: 12px 70px;
+    border-radius: 5px;
+    background-color: $--blue-color;
+    border: 1px solid $--blue-color;
+    color: #fff;
```
Comment (anastasiiavorobiova) [inline src/styles/cards.scss:90]: Consider using variables for repeating colors

File: src/styles/cards.scss [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+    transition-duration: 300ms;
+
+    &:hover {
+      border: 1px solid $--blue-color;
+      background-color: #fff;
```
Comment (anastasiiavorobiova) [inline src/styles/cards.scss:95]: Consider using variables for repeating colors

File: src/styles/main.scss (hunk 8 above)
Comment (anastasiiavorobiova) [reply src/styles/main.scss:8]: Not fixed

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

File: src/styles/main.scss (hunk 8 above)
Comment (anastasiiavorobiova) [reply src/styles/main.scss:8]: Not fixed

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! 
Screenshot 2023-11-20 at 15 51 24
The modifier change doesn't work 
Screenshot 2023-11-20 at 15 50 48
An active link should have a blue color. Feel free to ask for help in the chat

**Review by loralevitska — CHANGES_REQUESTED** [review]
Please redeploy your solution for review, test report link don't work
image

File: src/index.html [hunk 21]
```diff
@@ -5,8 +5,408 @@
...
+          alt="logo__MOYO"
+        >
+      </a>
+
+      <nav class="nav">
+        <ul class="nav__list">
+          <li class="nav__item">
+            <a
+              href="#"
+              class="nav__link"
```
Comment (maxim2310) [inline src/index.html:29]:
```suggestion
              class="nav__link is-active"
```

File: src/styles/main.scss (hunk 9 above)
Comment (maxim2310) [reply src/styles/main.scss:11]: https://github.com/mate-academy/layout_catalog/pull/3844/files#r1398232415
not fixed

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Good work? fix some changes and check why tests dont show relevant info

**Pull request closed without merging**